peps/pep-0751.rst

```diff
@@ -0,0 +1,901 @@
+  applicability is determined by ``package.marker``
+
+
+``package.direct``
+''''''''''''''''''
+
+- Optional
+- Boolean
+- Represents whether the installation is via a `direct URL reference`_
+
+
+``[[package.files]]``
+'''''''''''''''''''''
+
+- Must be specified if ``[package.vcs]`` is not
+- Array of tables
+- Tables can be written inline
+- Represents the files to potentially install for the package and version
+
+
+``package.files.name``
+''''''''''''''''''''''
+
+- String
+- The file name
+- Necessary for installers to decide what to install when using package locking
+
+
+``package.files.lock``
+''''''''''''''''''''''
+
+- Required when ``[[file-lock]]`` is used
+- Array of strings
+- An array of ``file-lock.name`` values which signify that the file is to be
+  installed when the corresponding ``[[file-lock]]`` table applies to the
+  environment
+- There MUST only be a single file with any one ``file-lock.name`` entry per
+  package, regardless of version
+
+
+``package.files.simple-repo-package-url``
+'''''''''''''''''''''''''''''''''''''''''
+
+- Optional (although mutually exclusive with
+  ``package.simple-repo-package-url``)
+- String
+- The value has the same meaning as ``package.simple-repo-package-url``
+- This key is avaible per-file to support :pep:`708` when some files override
+  what's provided by another `Simple Repository API`_ index
+
+
+``package.files.origin``
+''''''''''''''''''''''''
+
+- Optional
+- String
+- URI where the file was found when the lock file was generated
+- Useful for documenting where the file came from and potentially where to look
+  for the file if not already downloaded/available
+
+
+``package.files.hash``
+''''''''''''''''''''''
+
+- String
+- The hash value of the file contents using the hash algorithm specified by
+  ``hash-algorithm``
+- Used by installers to verify the file contents match what the locker worked
+  with
+
+
+``[package.vcs]``
+'''''''''''''''''
+
+- Must be specified if ``[[package.files]]`` is not (altough may be specified
+  simultaneously with ``[[package.files]]``)
+- Table representing the version control system containing the package and
+  version
+
+
+``package.vcs.type``
+''''''''''''''''''''
+
+- String
+- The type of version control system used
+- The valid values are specified by the
+  `registered VCSs <https://packaging.python.org/en/latest/specifications/direct-url-data-structure/#registered-vcs>`__
+  of the direct URL data structure
+
+
+``package.vcs.origin``
+''''''''''''''''''''''
+
+- String
+- The URI of where the repository was located when the lock file was generated
+
+
+``package.vcs.commit``
+''''''''''''''''''''''
+
+- String
+- The commit ID for the repository which represents the package and version
+- The value MUST be immutable for the VCS for security purposes
+  (e.g. no Git tags)
+
+
+``package.vcs.lock``
+'''''''''''''''''''''''''
+
+- Required when ``[[file-lock]]`` is used
+- An array of strings
```

Comment:
Recommend that they be sorted in lexicographic order?

Reply:
There's nothing to sort here; a package can only have one `package.vcs` entry.